### PR TITLE
Update to isort 5.8

### LINKED ISSUE #10287: Update to isort 5.7
I followed the setup instructions, and the default pre-commit hook fails as follows:
```bash
$ git commit --allow-empty

=============================================== FAILURES ===============================================
______________________________________________ test_isort ______________________________________________

    def test_isort() -> None:
        ''' Assures that the Python codebase imports are correctly sorted.
    
        '''
        chdir(TOP_PATH)
        proc = run(["isort", "-df", "-rc", "-c", *ls_files("*.py")], capture_output=True)
>       assert proc.returncode == 0, f"isort issues:\n{proc.stdout.decode('utf-8')}"
E       AssertionError: isort issues:
E         
E       assert 2 == 0
E        +  where 2 = CompletedProcess(args=['isort', '-df', '-rc', '-c', '_setup_support.py', 'bokeh/__init__.py', 'bokeh/__main__.py', 'bo...config] [--honor-noqa]\n             [files [files ...]]\nisort: error: argument -f/--future: expected one argument\n').returncode

tests/codebase/test_isort.py:35: AssertionError
======================================= short test summary info ========================================
FAILED tests/codebase/test_isort.py::test_isort - AssertionError: isort issues:
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/home/markus/bokeh/miniconda/envs/bkdev/lib/python3.8/subprocess.py:1011: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
===================================== 1 failed, 3 passed in 49.25s =====================================
^C
```

This seems to be caused by [isort 5 having changed its command line flags](https://timothycrosley.github.io/isort/docs/upgrade_guides/5.0.0/), and the test expecting isort v4.

I'd recommend the following changes
isort 4 | isort 5 | recommended action
-- | -- | --
`-df` `--diff` | `--df` `--diff` | change currently used `-df` to universally accepted `--diff`
`-rc` `--recursive` | no equivalent | remove `-rc` flag, looks like we pass individual files anyway.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #10287
- [x] tests added / passed

A few comments on goals:

* make all the individual `.isort.cfg` files as nearly identical as possible
    - `examples` and `sphinx` are identical, and a restricted to line length 88 since they need to be formatted in docs
    - `bokeh` and `release` are identical, and `test` almost so (it needs specual adds for local imports)
* I used our 165 length for `bokeh`, `release`, and `test` for consistency, but I *really* don't like super long import lines. I do not ever want to to have to scroll *horizontally* to see imports. I used `force_grid_wrap=5` which is not perfect, but seems OK

@mattpap we don't have to merge this especially soon, while so many PRs are in flight. But I did want to nail down a workable configuration (will keep looking for any tweaks to improve this as well)

### CODE REVIEW

File: bokeh/colors/hsl.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def to_rgb(self):
             :class:`~bokeh.colors.rgb.RGB`
 
         '''
-        from .rgb import RGB # prevent circular import
+        from .rgb import RGB  # prevent circular import
```
Comment (mattpap) [inline bokeh/colors/hsl.py:141]: This extra space is not a mistake?

File: bokeh/models/plots.py [hunk 2]
```diff
@@ -44,18 +44,8 @@
...
-    REQUIRED_SCALE,
-)
-from ..core.validation.warnings import (
-    FIXED_HEIGHT_POLICY,
-    FIXED_SIZING_MODE,
-    FIXED_WIDTH_POLICY,
-    MISSING_RENDERERS,
-)
+from ..core.validation.errors import BAD_EXTRA_RANGE_NAME, INCOMPATIBLE_SCALE_AND_RANGE, REQUIRED_RANGE, REQUIRED_SCALE
+from ..core.validation.warnings import FIXED_HEIGHT_POLICY, FIXED_SIZING_MODE, FIXED_WIDTH_POLICY, MISSING_RENDERERS
```
Comment (mattpap) [inline bokeh/models/plots.py:48]: What's the threshold for single vs multiple lines? Here the original was more readable, but then there are many other changes were a simple single line is blown out of proportion.

**Comment by mattpap** [comment]
> we don't have to merge this especially soon, while so many PRs are in flight.

Well, I would actually merge this sooner than later. I don't see too many PRs where this would be particularly conflicting.

File: bokeh/colors/hsl.py (hunk 1 above)
Comment (bryevdv) [reply bokeh/colors/hsl.py:141]: Not AFAIK, two spaces before a comment seems to be common in several tools at this point.

File: bokeh/models/plots.py (hunk 2 above)
Comment (bryevdv) [reply bokeh/models/plots.py:48]: So here are the options as far as I understand them

* Use `force_grid_wrap` to set the threshold at a *number of imports*. This is currently done, with value 5. We could lower it to 4 or even 2 if we just want it to always be mulit-line
* Instead, set the line-length to something lower, e.g. 100, tho this seems weird to be contradicting with our other linter, and can lead to ~bad results (IMO) on indented imports  
* There is also evidently a "wrap length" which can be lower than the line length, but I had mixed results with it in brief experiments. I could experiment a bit more with it.

Comment (bryevdv) [reply bokeh/models/plots.py:48]: I can't get `wrap_length` to do anything I understand. I think the best option is `force_grid_wrap`, do you have any thoughts on a different value?

**Comment by bryevdv** [comment]
@mattpap I've pushed a  change with a wrap limit of 4 imports you can see if you like better.

**Comment by bryevdv** [comment]
I personally think the new value (4) is better

File: bokeh/util/string.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 '''
 
+# Standard library imports
```
Comment (mattpap) [inline bokeh/util/string.py:11]:
```suggestion
```

File: examples/app/gapminder/data.py [hunk 4]
```diff
@@ -2,7 +2,8 @@
 
 
 def process_data():
-    from bokeh.sampledata.gapminder import fertility, life_expectancy, population, regions
+    from bokeh.sampledata.gapminder import (fertility, life_expectancy,
+                                            population, regions)
```
Comment (mattpap) [inline examples/app/gapminder/data.py:6]: Why are we using different formatting in examples? This is also the worst way, in my opinion, to format imports (and data structures for that matter), as results in this:
```py
from some.very.very.very.very.long.module.path import (a,
                                                       b,
                                                       c)
```

Comment (bryevdv) [reply examples/app/gapminder/data.py:6]: The docs and example are different in order that they may be more compact. While what you show is technically a possibility, I have not actually seen what you describe ever be a problem in actual practice in our specific repo. 

cc @tcmetzger do you have any thoughts about import formatting in docs code examples? The available isort multi line output options are described here: https://pycqa.github.io/isort/docs/configuration/multi_line_output_modes/

**Comment by bryevdv** [comment]
FYI I did have some conflicts rebasing this on latest, so I'd like to go ahead  and merge this as-is and punt on any question or the docs/examples (if it is really a concern, an issue can be opened).

**Pull request merged**